### PR TITLE
drivers: adc: stm32: allow overrun

### PR DESCRIPTION
**Pull request by JvanDooren**
Allow overrun to happen and continue sampling

### CODE REVIEW

File: drivers/adc/adc_stm32.c [hunk 1]
```diff
@@ -628,6 +639,7 @@ static int start_read(const struct device *dev,
 			}
 #else
 			LL_ADC_REG_SetSequencerRanks(adc, table_rank[data->channel_count], channel);
+			LL_ADC_REG_SetSequencerLength(adc, table_seq_len[data->channel_count]);
```
Comment (aborowski-msol) [inline drivers/adc/adc_stm32.c:637]: IMO makes no sense to set sequence length in for loop. Set only once as number of channels is determined (line 644)

File: drivers/adc/adc_stm32.c [hunk 2]
```diff
@@ -262,33 +262,44 @@ struct adc_stm32_cfg {
...
 
-
-		/* Stop the DMA engine, only to start it again when the callback returns
-		 *  ADC_ACTION_REPEAT or ADC_ACTION_CONTINUE
-		 *  and the number of samples haven't been reached
-		 *  Starting the DMA engine is done witthin adc_context_start_sampling
-		 */
-		dma_stop(data->dma.dma_dev, data->dma.channel);
-		adc_context_on_sampling_done(&data->ctx, dev);
+		if (LL_ADC_IsActiveFlag_OVR(adc)) {
```
Comment (aborowski-msol) [inline drivers/adc/adc_stm32.c]: handled same way as normal case. woudn't just:
if (LL_ADC_IsActiveFlag_OVR(adc)) { LL_ADC_ClearFlag_OVR(adc); }
do the trick in normal flow?

File: drivers/adc/adc_stm32.c (hunk 1 above)
Comment (JvanDooren) [reply drivers/adc/adc_stm32.c:637]: Well, this is exactly what LL_ADC_REG_SetSequencerRanks also does, and in this case we don't need the -1 to correct the index.

Comment (JvanDooren) [reply drivers/adc/adc_stm32.c:637]: And, it's also what this guy did: https://github.com/pdietl/zephyr/commit/b50dc4564f11aad36df5925d7a13b405c817a600#diff-14bcedc892b7643865d068e532c3dab525465a9814a505eba6724db5026028b7

File: drivers/adc/adc_stm32.c (hunk 2 above)
Comment (JvanDooren) [reply drivers/adc/adc_stm32.c]: You mean combine the normal flow and the handling of the overflow code?

Comment (aborowski-msol) [reply drivers/adc/adc_stm32.c]: Yes, as the only difference is only clearing the flag if it is set

File: drivers/adc/adc_stm32.c (hunk 1 above)
Comment (aborowski-msol) [reply drivers/adc/adc_stm32.c:637]: Well... LL_ADC_REG_SetSequencerRanks is called per channel, so it is in the right place. But setting length each time... I think not the best way, but this is just an opinion. This way would also work.

Comment (JvanDooren) [reply drivers/adc/adc_stm32.c:637]: Indeed, I'd rather leave it as is.

File: drivers/adc/adc_stm32.c (hunk 2 above)
Comment (JvanDooren) [reply drivers/adc/adc_stm32.c]: Fixed

**Review by aborowski-msol — APPROVED** [review]

**Pull request merged**